### PR TITLE
chore: update otel-collector-config.yaml

### PR DESCRIPTION
**Pull request by MulL1on**
#### What type of PR is this?

chore: update otel-collector-config.yaml

en: Jaeger now supports OTLP directly, the jaeger export was deprecated in the collector.  Otel-collector now fails to decode jaeger exporter.

### CODE REVIEW

File: opentelemetry/docker-compose.yaml [hunk 1]
```diff
@@ -27,6 +27,7 @@ services:
       - "14268"
       - "14250:14250"
```
Comment (CoderPoet) [inline opentelemetry/docker-compose.yaml:28]: This can be removed

File: opentelemetry/server/main.go [hunk 2]
```diff
@@ -58,7 +58,7 @@ func main() {
 
 	p := provider.NewOpenTelemetryProvider(
 		provider.WithServiceName(serviceName),
-		provider.WithExportEndpoint("host.docker.internal:4317"),
+		provider.WithExportEndpoint(":4317"),
```
Comment (CoderPoet) [inline opentelemetry/server/main.go:61]: Can it be set to `jaeger-all-in-one:4317` ? Or put it in an environment variable?

**Review by CoderPoet — APPROVED** [review]
/LGTM

**Comment by MulL1on** [comment]
The configuration for the client-side exporter appears to be incorrect. “host.docker.internal” is only available on Windows and macOS.

**Review by li-jin-gou — APPROVED** [review]

**Pull request merged**